Onboarding/QA/tests/profile-generic_test.exs

```diff
@@ -0,0 +1,263 @@
+    |> FlowTester.set_global_dict("config", %{"contentrepo_token" => auth_token})
+  end
+
+  # This lets us have cleaner button/list assertions.
+  def indexed_list(var, labels) do
+    Enum.with_index(labels, fn lbl, idx -> {"@#{var}[#{idx}]", lbl} end)
+  end
+
+  # The common case for buttons.
+  defmacro button_labels(labels) do
+    quote do: unquote(indexed_list("button_labels", labels))
+  end
+
+  # The common case for lists.
+  defmacro list_items(labels) do
+    quote do: unquote(indexed_list("list_items", labels))
+  end
+
+  describe "profile generic" do
+    test "30% complete" do
+      setup_flow()
+      |> FlowTester.set_contact_properties(%{"year_of_birth" => "1988", "province" => "Western Cape", "area_type" => "", "gender" => "male"}) # Basic Information
+      |> FlowTester.set_contact_properties(%{"relationship_status" => "", "education" => "", "socio_economic" => "", "other_children" => ""}) # Personal Information
+      |> FlowTester.set_contact_properties(%{"dma_01" => "answer", "dma_02" => "", "dma_03" => "", "dma_04" => "", "dma_05" => ""}) # Daily Life
+      |> FlowTester.start()
+      |> fn step ->
+        [msg] = step.messages
+        assert String.contains?(msg.text, "Basic information 3/4")
+        assert String.contains?(msg.text, "Personal information 0/4")
+        assert String.contains?(msg.text, "Daily life 1/5")
+        step
+      end.()
+      |> receive_message(%{
+        text: "Your profile is already 30% complete" <> _,
+        buttons: button_labels(["Continue", "Why?"])
+      })
+    end
+
+    test "30% complete -> why -> let's go" do
+      setup_flow()
+      |> FlowTester.set_contact_properties(%{"year_of_birth" => "1988", "province" => "Western Cape", "area_type" => "", "gender" => "male"}) # Basic Information
+      |> FlowTester.set_contact_properties(%{"relationship_status" => "", "education" => "", "socio_economic" => "", "other_children" => ""}) # Personal Information
+      |> FlowTester.set_contact_properties(%{"dma_01" => "answer", "dma_02" => "answer2",}) # Daily Life
+      |> FlowTester.start()
+      |> fn step ->
+        [msg] = step.messages
+        assert String.contains?(msg.text, "Basic information 3/4")
+        assert String.contains?(msg.text, "Personal information 0/4")
+        assert String.contains?(msg.text, "Daily life 2/5")
+        step
+      end.()
+      |> receive_message(%{
+        text: "Your profile is already 30% complete" <> _,
+        buttons: button_labels(["Continue", "Why?"])
+      })
+      |> FlowTester.send(button_label: "Why?")
+      |> receive_message(%{
+        text: "ℹ️ Our team of experts has put together" <> _,
+        buttons: button_labels(["Yes, let's go", "Not right now"])
+      })
+      |> FlowTester.send(button_label: "Yes, let's go")
+      |> receive_message(%{
+        text: "🟩🟩🟩🟩🟩🟩🟩🟩\r\n\r\nYour profile is 100% complete" <> _,
```

Comment:
Some messages use `\r\n`, others use `\n`. I think this is something we need to address in contentrepo, because the inconsistency makes testing difficult and may cause other problems at some point.

Reply:
I did try to match them for now with the content repo messages, so if you use CMS or fakeCMS, you should get the same result.